src/main/java/frc/robot/subsystems/Hood.java

```diff
@@ -81,6 +81,9 @@ private double getRawAngle() {
    */
   @Override
   public void setSetpoint(double setpoint) {
+    if (getController() != null) {
+      super.setSetpoint(Utilities.clip(setpoint, Constants.Hood.MIN_ANGLE, Constants.Hood.MAX_ANGLE));
+    }
     super.setSetpoint(Utilities.clip(setpoint, Constants.Hood.MIN_ANGLE, Constants.Hood.MAX_ANGLE));
```

Comment:
This doesn't make sense. You've got the super.setSetpoin wrapped in an if, but then you've also got it right afterwards.